pkg/resource/subnet_group/hooks.go

```diff
@@ -0,0 +1,142 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package subnet_group
+
+import (
+	"context"
+
+	svcapitypes "github.com/aws-controllers-k8s/memorydb-controller/apis/v1alpha1"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+)
+
+// getTags gets tags from given ParameterGroup.
+func (rm *resourceManager) getTags(
```

Comment:
Since this would same across different resources can we use it from a common place?

Reply:
Good point. This is a little bit tricky but feasible. e.g https://github.com/aws-controllers-k8s/sfn-controller/blob/main/pkg/util/tags.go#L28-L67
However if you want to move utility functions to a common place, please do it in a separate PR, to make it easy to review